xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocumentArchive.java

```diff
@@ -453,30 +478,47 @@ public Version getPrevVersion(Version ver)
     }
 
     /**
-     * @param ver - for what version find nearest
-     * @return nearest version which contain full information (not patch)
+     * @deprecated This method has been deprecated as its name is error-prone, and it has never done what's documented:
+     * it always returned the next full version, and never the nearest one. So we renamed it to
+     * {@link #getNextFullVersion(Version)}.
+     *
+     * @param ver - for what version find next full one
+     * @return next version which contain full information (not patch)
      */
+    @Deprecated(since = "14.10.12,15.5RC1")
     public Version getNearestFullVersion(Version ver)
```

Comment:
Is this method still used somewhere?

Reply:
To my knowledge no but was a bit lazy to create the aspect just for it...

Reply:
yeah, I can related :)